modules/Vtiger/inventoryfields/Reference.php

```diff
@@ -83,8 +93,14 @@ public function getReferenceModule($record)
 	{
 		if (!empty($record)) {
 			$metadata = vtlib\Functions::getCRMRecordMetadata($record);
-
-			return $metadata['setype'];
+			$referenceModuleList = $this->getReferenceModules();
+			$referenceEntityType = $metadata['setype'];
+			if (!empty($referenceModuleList) && \in_array($referenceEntityType, $referenceModuleList)) {
+				return Vtiger_Module_Model::getInstance($referenceEntityType);
+			}
+			if (!empty($referenceModuleList) && \in_array('Users', $referenceModuleList)) {
+				return Vtiger_Module_Model::getInstance('Users');
+			}
```

Comment:
You cannot assume that if the record is not found, it means that it is from users.

Reply:
The whole pull-request has a wrong assumption.
If you want to be associated with a user, create a new field type.

Reply:
MMm i have do the same behavior of non-inventory reference field for consistency. see : https://github.com/YetiForceCompany/YetiForceCRM/blob/439d5fe68a0ffaccda9189ecce66d6f55a27d345/modules/Vtiger/uitypes/Reference.php#L64

Reply:
MMm i have do the same behavior of non-inventory reference field for consistency. see : https://github.com/YetiForceCompany/YetiForceCRM/blob/439d5fe68a0ffaccda9189ecce66d6f55a27d345/modules/Vtiger/uitypes/Reference.php#L64